src/main/java/spireMapOverhaul/zones/ambushed/AmbushedZone.java

```diff
@@ -235,6 +235,43 @@ private boolean isCardForShop(String cardID, AbstractCard.CardRarity rarity, Abs
         return card != null && card.rarity == rarity && card.type == type && isCardColored == isColored;
     }
 
+    @Override
+    public AbstractCard getReplacementShopCardForCourier(AbstractCard purchasedCard) {
+        boolean isColored = purchasedCard.color != AbstractCard.CardColor.COLORLESS;
+        AbstractCard replacementCard = getCardForCourier(purchasedCard.type, isColored);
+
+        // Handle the case where no replacement card is found
+        if (replacementCard == null) {
+            // You can decide how to handle this - return null, a default card, etc.
```

Comment:
Please remove the default comments if you don't need them to explain complex beahvior